### ISSUE TITLE
Rule Proposal: constructor-super

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#2720]
If the class declaration has `extends`, its constructor should have `super()` calling.

The following patterns are considered warnings:

```js
class A extends B {
  constructor() { }
}
```

The following patterns are not considered warnings:

```js
class A extends B {
  constructor() {
    super();
  }
}
```

```js
class A {
  constructor() { }
}
```

```js
class A extends null {
  constructor() { }
}
```

I'm happy to work if it was accepted.

**Comment by michaelficarra** [comment]
Like https://github.com/eslint/eslint/issues/2721, this is not something that we can statically determine. :-1:

**Comment by nzakas** [comment]
@michaelficarra I think what you're saying is that `super()` must be called during the constructor, but not necessarily directly within `constructor`?  If not, can you explain why this can't be statically determined?

**Comment by michaelficarra** [comment]
@nzakas I was referring to `super()` that would be conditionally called. But if we're okay with the following program not producing a warning, I'm good with this.

```js
class A extends B {
  constructor() {
    if (false) super();
  }
}
```

**Comment by nzakas** [comment]
Yeah, I think we are OK with that.
